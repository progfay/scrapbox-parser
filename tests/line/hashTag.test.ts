/* global describe it expect */

import '../jest-setup'

describe('hashTag', () => {
  it('Simple hashTag', () => {
    expect('#tag').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'hashTag',
            href: 'tag'
          }
        ]
      }
    ])
  })

  it('Only `#` is not hashTag', () => {
    expect('#').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'plain',
            text: '#'
          }
        ]
      }
    ])
  })

  it('HashTag includes `#`', () => {
    expect('#hash#Tag').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'hashTag',
            href: 'hash#Tag'
          }
        ]
      }
    ])
  })

  it('HashTag in sentence with spaces', () => {
    expect('This is a #tag .').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'plain',
            text: 'This is a '
          },
          {
            type: 'hashTag',
            href: 'tag'
          },
          {
            type: 'plain',
            text: ' .'
          }
        ]
      }
    ])
  })

  it('HashTag in sentence without spaces is not hashTag', () => {
    expect('→#notTag←').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'plain',
            text: '→#notTag←'
          }
        ]
      }
    ])
  })

  it('Multiple hashTag', () => {
    expect('#hoge #fuga #piyo').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'hashTag',
            href: 'hoge'
          },
          {
            type: 'plain',
            text: ' '
          },
          {
            type: 'hashTag',
            href: 'fuga'
          },
          {
            type: 'plain',
            text: ' '
          },
          {
            type: 'hashTag',
            href: 'piyo'
          }
        ]
      }
    ])
  })
})
