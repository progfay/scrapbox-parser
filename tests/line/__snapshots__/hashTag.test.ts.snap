// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hashTag HashTag in sentence with spaces 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "raw": "This is a",
        "text": "This is a",
        "type": "plain",
      },
      {
        "raw": " ",
        "text": " ",
        "type": "plain",
      },
      {
        "href": "tag",
        "raw": "#tag",
        "type": "hashTag",
      },
      {
        "raw": " .",
        "text": " .",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;

exports[`hashTag HashTag in sentence without spaces is not hashTag 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "raw": "→#notTag←",
        "text": "→#notTag←",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;

exports[`hashTag HashTag includes \`#\` 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "href": "hash#Tag",
        "raw": "#hash#Tag",
        "type": "hashTag",
      },
    ],
    "type": "line",
  },
]
`;

exports[`hashTag Multiple hashTag 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "href": "hoge",
        "raw": "#hoge",
        "type": "hashTag",
      },
      {
        "raw": " ",
        "text": " ",
        "type": "plain",
      },
      {
        "href": "fuga",
        "raw": "#fuga",
        "type": "hashTag",
      },
      {
        "raw": " ",
        "text": " ",
        "type": "plain",
      },
      {
        "href": "piyo",
        "raw": "#piyo",
        "type": "hashTag",
      },
    ],
    "type": "line",
  },
]
`;

exports[`hashTag Only \`#\` is not hashTag 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "raw": "#",
        "text": "#",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;

exports[`hashTag Simple hashTag 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "href": "tag",
        "raw": "#tag",
        "type": "hashTag",
      },
    ],
    "type": "line",
  },
]
`;
